src/notifier/mqtt_notifier.py

```diff
@@ -0,0 +1,121 @@
+# std
+import json
+import logging
+from typing import List
+
+# project
+from . import Notifier, Event
+
+
+class MqttNotifier(Notifier):
+    def __init__(self, title_prefix: str, config: dict):
+        logging.info("Initializing MQTT notifier.")
+        super().__init__(title_prefix, config)
+
+        self._username = None
+        self._password = None
+
+        try:
+
+            credentials = config["credentials"]
+            self._set_config(config)
+            self._set_credentials(credentials)
+
+        except KeyError as key:
+            logging.error(f"Invalid config.yaml. Missing key: {key}")
+
+        self._init_mqtt()
+
+    def _set_config(self, config: dict):
+
+        self._topic = config["topic"]
```

Comment:
Nitpick: I think it'd be cleaner if you either catch raised KeyError exceptions directly where they happen or if the function is expected to raise such exception add a doctoring with `:raises KeyError: If a key in the config doesn't exist`. Makes it a bit more explicit that these functions need to be called inside try-except.

Reply:
Used the suggested docstring since I would prefer not to duplicate the exception handling 3 times.